pkg/controllermanager/controller/shoot/reference/reconciler.go

```diff
@@ -155,29 +158,44 @@ func (r *Reconciler) handleReferencedSecrets(ctx context.Context, log logr.Logge
 			return nil
 		})
 	}
-	err := flow.Parallel(fns...)(ctx)
 
-	return added != 0, err
+	return added != 0, flow.Parallel(fns...)(ctx)
 }
 
 func (r *Reconciler) handleReferencedConfigMap(ctx context.Context, log logr.Logger, c client.Client, shoot *gardencorev1beta1.Shoot) (bool, error) {
+	var (
+		fns            []flow.TaskFn
+		added          = uint32(0)
+		configMapNames = namesForReferencedResources(shoot, "ConfigMap")
+	)
+
 	if configMapRef := getAuditPolicyConfigMapRef(shoot.Spec.Kubernetes.KubeAPIServer); configMapRef != nil {
-		configMap := &corev1.ConfigMap{}
-		if err := c.Get(ctx, kubernetesutils.Key(shoot.Namespace, configMapRef.Name), configMap); err != nil {
-			return false, err
-		}
+		configMapNames = append(configMapNames, configMapRef.Name)
+	}
 
-		if !controllerutil.ContainsFinalizer(configMap, FinalizerName) {
-			log.Info("Adding finalizer to ConfigMap", "configMap", client.ObjectKeyFromObject(configMap))
-			if err := controllerutils.AddFinalizers(ctx, r.Client, configMap, FinalizerName); err != nil {
-				return true, fmt.Errorf("failed to add finalizer to ConfigMap: %w", err)
+	for _, configMapName := range configMapNames {
+		name := configMapName
+		fns = append(fns, func(ctx context.Context) error {
+			configMap := &corev1.ConfigMap{}
+			s := shoot
+			if err := c.Get(ctx, kubernetesutils.Key(s.Namespace, name), configMap); err != nil {
```

Comment:
```suggestion
			if err := c.Get(ctx, kubernetesutils.Key(shoot.Namespace, name), configMap); err != nil {
```
I think you can drop this.